web3/contract/base_contract.py

```diff
@@ -258,8 +272,12 @@ def _get_event_filter_params(
         return event_filter_params
 
     @classmethod
-    def factory(cls, class_name: str, **kwargs: Any) -> PropertyCheckingFactory:
-        return PropertyCheckingFactory(class_name, (cls,), kwargs)
+    def factory(
+        cls, class_name: str, **kwargs: Any
+    ) -> Union["ContractEvent", "AsyncContractEvent"]:
```

Comment:
I wonder if this should be `TContractEvent`?

Reply:
I found that using `TContractEvent` requires one argument for `factory` to be of that same type.

Reply:
Ah, okay. Thanks for looking into it!